docs/ai/python.rst

```diff
@@ -0,0 +1,414 @@
+.. _ref_ai_python:
+
+======
+Python
+======
+
+:edb-alt-title: EdgeDB AI's Python package
+
+The ``edgedb.ai`` package is an optional binding of the AI extension in EdgeDB.
+To use the AI binding, you need to install ``edgedb-python`` with the ``ai``
+extra dependencies:
+
+.. code-block:: bash
+
+  $ pip install 'edgedb[ai]'
+
+
+Usage
+=====
+
+Start by importing ``edgedb`` and ``edgedb.ai``:
+
+.. code-block:: python
+
+    import edgedb
+    import edgedb.ai
+
+
+Blocking
+--------
+
+The AI binding is built on top of the regular EdgeDB client objects, providing
+both blocking and asynchronous versions of its API. For example, a blocking AI
+client is initialized like this:
+
+.. code-block:: python
+
+    client = edgedb.create_client()
+    gpt4ai = edgedb.ai.create_ai(
+        client,
+        model="gpt-4-turbo-preview"
+    )
+
+Add your query as context:
+
+.. code-block:: python
+
+    astronomy_ai = gpt4ai.with_context(
+        query="Astronomy"
+    )
+
+The default text generation prompt will ask your selected provider to limit
+answer to information provided in the context and will pass the queried
+objects' AI index as context along with that prompt.
+
+Call your AI client's ``query_rag`` method, passing in a text query.
+
+.. code-block:: python
+
+    print(
+        astronomy_ai.query_rag("What color is the sky on Mars?")
+    );
+
+or stream back the results by using ``stream_rag`` instead:
+
+.. code-block:: python
+
+    for data in astronomy_ai.stream_rag("What color is the sky on Mars?"):
+        print(data)
+
+
+Async
+-----
+
+To use an async client instead, do this:
+
+.. code-block:: python
+
+    import asyncio  # alongside the EdgeDB imports
+
+    client = edgedb.create_async_client()
+
+    async def main():
+        gpt4ai = await edgedb.ai.create_async_ai(
+            client,
+            model="gpt-4-turbo-preview"
+        )
+        astronomy_ai = gpt4ai.with_context(
+            query="Astronomy"
+        )
+        query = "What color is the sky on Mars?"
+        print(
+            await astronomy_ai.query_rag(query)
+        );
+
+        #or streamed
+        async for data in blog_ai.stream_rag(query):
+            print(data)
+
+    asyncio.run(main())
+
+
+API Reference
+=============
+
+.. py:function:: create_ai(client, **kwargs) -> EdgeDBAI
+
+   Creates an instance of ``EdgeDBAI`` with the specified client and options.
+
+   This function ensures that the client is connected before initializing the
+   AI with the specified options.
+
+   :param client:
+       An EdgeDB client instance.
+
+   :param kwargs:
+       Keyword arguments that are passed to the ``AIOptions`` data class to
+       configure AI-specific options. These options are:
+
+       * ``model``: The name of the model to be used. (required)
+       * ``prompt``: An optional prompt to guide the model's behavior. (default: None)
```

Comment:
I _think_ setting this to `None` explicitly or via default means you end up using the system prompt? This description is strictly true, but probably misleading as to the end result.

Reply:
I added an explanation here of what the result of `None` actually is.